### PR TITLE
Fix conflict when granting permissions for an existent team

### PR DESCRIPTION
**Pull request by mauricioabreu**

### CODE REVIEW

File: internal/provider/resource_tsuru_app_grant.go [hunk 1]
```diff
@@ -53,7 +54,11 @@ func resourceTsuruApplicationGrantCreate(ctx context.Context, d *schema.Resource
 	team := d.Get("team").(string)
 
 	err := resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
-		_, err := provider.TsuruClient.AppApi.AppTeamGrant(ctx, app, team)
+		response, err := provider.TsuruClient.AppApi.AppTeamGrant(ctx, app, team)
+		// ignore teams already granted for this app
+		if response.StatusCode == http.StatusConflict {
+			return nil
+		}
```
Comment (nettoclaudio) [inline internal/provider/resource_tsuru_app_grant.go:61]: I'd combined this validation with an extra one against the message error which Tsuru returns - "team already have access to this app". A simple `strings.Contains` in the error message should cover it.

**Review by nettoclaudio — CHANGES_REQUESTED** [review]
Just a nitpick to make clear the context in the future.

File: internal/provider/resource_tsuru_app_grant.go (hunk 1 above)
Comment (mauricioabreu) [reply internal/provider/resource_tsuru_app_grant.go:61]: Thanks for the tips!
What would happen if the tsuru's upstream API changes the error message? Would it break?

Comment (nettoclaudio) [reply internal/provider/resource_tsuru_app_grant.go:61]: Probably... I agree, it's a tradeoff. I took a look on Tsuru code and realized that it only returns 409 in this condition (https://github.com/tsuru/tsuru/blob/main/api/app.go#L869)... so I think it's safe go for it.

**Review by nettoclaudio — APPROVED** [review]
lgtm

**Pull request merged**